spyder/plugins/completion/kite/status.py

```diff
@@ -0,0 +1,45 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright © Spyder Project Contributors
+# Licensed under the terms of the MIT License
+# (see spyder/__init__.py for details)
+
+"""
+Status widget for Kite completions.
+"""
+
+# Local imports
+from spyder.config.base import _
+from spyder.utils import icon_manager as ima
+from spyder.widgets.status import BaseTimerStatus
+
+
+class KiteStatus(BaseTimerStatus):
+    """Status bar widget for Kite completions status."""
+
+    def __init__(self, parent, statusbar):
+        super(KiteStatus, self).__init__(parent, statusbar,
+                                         icon=ima.get_kite_icon())
+
+    def import_test(self):
+        """Raise ImportError if feature is not supported."""
+        try:
+            from spyder.plugins.completion.kite.plugin import (
+                KiteCompletionPlugin)
+            KiteCompletionPlugin.kite_status()
```

Comment:
Ok, I noticed that we're not saving `parent` in `StatusBarWidget` (the base class of `BaseTimerStatus`), but I think we could do that to avoid adding and calling class methods.

@goanpeca, what do you think?

Reply:
I think we would need to use a new argument then since `parent` needs to be a `QWidget` (`KiteCompletionPlugin` isn't a QWidget), and I can add that to the `KiteStatus` subclass

Reply:
Maybe we can have `main` and `parent` as the parameters of the constructor and have the parent stored as `self._parent` and main as `self._main` (or calling it as the actual parent in the super method